docker/justfile

```diff
@@ -53,6 +53,13 @@ test: test-js test-py
 
 # run dev server in docker container
 serve env="dev" *args="":
+    #!/usr/bin/env bash
+    if [ "$env" = "prod" ]; then
+        echo foo
+        SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+        mkdir "$SCRIPT_DIR/storage"
+        sudo chown 10003:10003 "$SCRIPT_DIR/storage"
+    fi
     docker-compose up {{ args }} {{ env }}
```

Comment:
some suggestions for justfile style - you have to use bash in order to do the test, but then you have this thing that the behaviour subtly changes if one command fails in a justfile recipe. The just default is I think pretty much `pipefail` (i.e. if one command fails, the whole recipe fails), but when you switch to bash it switches to the bash default (i.e. if one command fails, the recipe carries on). So, at the moment we're pretty much adding  `set -euo pipefail` anytime we have `#!/usr/bin/env bash`, like this:

```bash
serve env="dev" *args="":
    #!/usr/bin/env bash
    set -euo pipefail
    if [ "$env" = "prod" ]; then
    ...etc...
```

The other thing is a purely optional suggestion, which is to use a [private recipe](https://just.systems/man/en/chapter_49.html) & [pass the variables as arguments to the recipe](https://github.com/casey/just#recipe-parameters), which would look something like this:

```suggestion
_create_storage env:
    #!/usr/bin/env bash
    set -euo pipefail

    if [ "$env" = "prod" ]; then
        echo foo
        SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
        mkdir "$SCRIPT_DIR/storage"
        sudo chown 10003:10003 "$SCRIPT_DIR/storage"
    fi

serve env="dev" *args="": (_create_storage env)
    docker-compose up {{ args }} {{ env }}
```